numpy/core/src/multiarray/mapping.c

```diff
@@ -3197,6 +3259,21 @@ PyArray_MapIterArray(PyArrayObject * a, PyObject * index)
 }
 
 
+/*NUMPY_API
+ *
+ * Use advanced indexing to iterate an array. Please note
```

Comment:
This needs to be documented somewhere public. It seems dangerous to expose an API that may change...